### ISSUE TITLE
False positive in RuboCop::Cop::Performance::HashEachMethods

### ISSUE THREAD

**Issue by alex-damian-negru** [rubocop-hq/rubocop#9265]
I am using Dry::Types, which allows me to build enum objects, such as this:

```ruby
Level = Dry::Types::String.enum('basic', 'active', 'advanced')
```

The resulting object has a `#values` method which yields an `Array` containing the values above. 

Using `Level.values.each` triggers this cop, even though I'm not operating on a hash.

--------

## Expected behavior

No offense should be added when using `#values` on objects other than those belonging to `Hash`.

## Actual behavior

Offense was added for a class different from `Hash`.

## Steps to reproduce the problem

This should do the trick:

```ruby
class Whatever
  def initialize(*items)
    @items = items
  end

  def values
    @items
  end
end

Whatever.new('a', 'b', 'c').values.each { |item| p item }
```

## RuboCop version

```
1.6.1 (using Parser 2.7.2.0, rubocop-ast 1.3.0, running on ruby 2.6.6 x86_64-linux-musl)
  - rubocop-performance 1.9.1
  - rubocop-rspec 2.1.0
```
